### ISSUE TITLE
feat(prompts): externalizar few-shots de dominio para YAML

### ISSUE THREAD

**Issue by ricmed** [ricmed/zettel_app#157]
## Contexto

A escala de relevância 1-5 de `prompts/literature_note.md:118-146` está ancorada em **cinco exemplos de Machine Learning** ("Machine learning é usado em IA", "Redes neurais têm camadas de neurônios", "Normalização Batch reduz covariate shift", "Dropout funciona como ensemble implícito", "Attention permite modelagem de dependências de longo alcance").

Essa escala alimenta `extraction.min_relevance_score` e **50% de `review_confidence`** (`extractor._score_review_confidence`, ADR-017). Com âncoras de ML, um argumento de Kant não tem régua para ser pontuado — o modelo cai no genérico. O mesmo vale, em menor grau, para os exemplos de tese, tags, rejeição e o exemplo de chunk aceito (RAG inteiro).

## Escopo

**Novo `zettel/domain_examples.py`** — espelhar `zettel/taxonomy.py`: modelos Pydantic + loader + renderizadores markdown, só `yaml` + `pydantic` (módulo folha, não pode arrastar chromadb). API: `load_domain_examples(path)` com erro tipado `DomainExamplesLoadError`, e `render_for_prompt(examples, prompt_name) -> dict[str, str]`.

**Novo `config/domain_examples.yaml`** — um arquivo só, com exemplos dos **dois** domínios. Isso é deliberado: um "nível 5" de ML ao lado de um "nível 5" de Filosofia ensina que a escala mede densidade conceitual, não proximidade com ML.

```
literature_note:
  relevance_examples / thesis_examples / judgement_examples
  tag_examples / rejection_examples / accepted_example
permanent_note:
  thesis_examples / decision_examples
```

**Placeholders em `prompts/literature_note.md`** (6): linhas 118-146 (os cinco `**Exemplo**:` viram um `{relevance_examples}` consolidado após a escala), 165-171 → `{thesis_examples}`, 193-218 → `{judgement_examples}`, 234-241 → `{tag_examples}`, 342-370 → `{rejection_examples}`, 374-402 → `{accepted_example}`.

**`prompts/permanent_note.md`** (2 placeholders + 3 neutralizações inline): 78-86 → `{thesis_examples}`, 254-284 → `{decision_examples}`; inline nas linhas 51, 21 e 30-32.

**Prompts de artigo** (2 neutralizações inline, sem placeholder): `article_section_blog.md:14-15` tem um livro real de ML hardcoded ("Alessandro Negro, *Knowledge Graphs and LLMs in Action*") e `article_section_academic.md:16` tem `(NEGRO et al., 2026)`. Ilustram formato de citação, não julgamento — trocar por autor/título genérico.

**Generalizar `fragmented`** (3 linhas, sem parametrizar): o enum `RejectionCategory` (`schemas.py:17`) é persistido em `summary_json` e lido por scripts de calibração — variar valores por domínio destruiria a comparabilidade. Filosofia tem o análogo exato de "fragmento sem interpretação": notação lógica sem glosa, citação em grego/latim sem tradução, tabela de silogismos. Editar `literature_note.md:61`, `:338` e `:272`.

## Duas armadilhas

**1. `fill_template` deixa a chave desconhecida LITERAL no prompt** (`zettel/llm.py:466-471`), não substitui por vazio. O renderizador tem de devolver **todas** as chaves sempre, mesmo com a seção ausente no YAML (string vazia), senão um `{relevance_examples}` cru vaza para o LLM.

**2. Contrato AST nos callers.** `tests/test_prompts.py:117-139` (`_caller_mapping_keys`) lê o literal `mapping = {...}` de `extractor.py:254` e `connector.py:354` via AST, e `test_placeholders_match_caller_mapping` (`:199-212`) exige **igualdade bidirecional**. Um `mapping.update(examples)` ou `**examples` é invisível ao AST e **quebra o teste** — as chaves têm de aparecer explicitamente no literal. Carregar os exemplos **uma vez por run** (ao lado do `load_prompt_parts` em `extractor.py:94` / `connector.py:184`), nunca por chunk.

## Testes

- `zettel/cli/diagnostics.py`: checagem do arquivo de exemplos, espelhando a de taxonomia em `diagnostics.py:251-270`.
- Estender `tests/test_prompts.py:242-246` (prompt declara as chaves) e `tests/test_extractor.py:373` (caller preenche, nada sobra no prompt). São complementares.
- **Cobertura que se move:** `test_json_examples_are_valid_json` valida os blocos ```json dos prompts. Ao mover o exemplo de chunk aceito para o YAML esse bloco some e o teste passa a vazio. Compensar validando `accepted_example` no loader contra `LiteratureChunkOutput` (`zettel/schemas.py`) — validação mais forte, reusando modelo existente.
- Novo: loader com arquivo ausente e com seção ausente (deve dar string vazia, nunca chave crua).

## Aceite

`grep -rniE "machine learning|redes neurais|tensorflow|dropout|gradient|overfit|covariate|regularizacao|series temporais" prompts/` retorna vazio (ou só `ptbr_guard.md:9`, benigno).

Depende de: T1.

**Comment by ricmed** [comment]
Duas adições ao escopo de documentação, levantadas depois de abrir a issue.

**1. `CLAUDE.md` é uma quarta superfície de acoplamento**, além de config, prompt e thresholds. Ele não só descreve a arquitetura: ele **trava decisões em prosa normativa**. Dois trechos ficam desatualizados com esta issue e precisam ser revisados junto:

- `CLAUDE.md:56` descreve a escala de relevância e a justificativa do `_RELEVANCE_FLOOR_CREDIT` citando a redação do prompt de extract. Se os few-shots saem para YAML, a descrição do "onde isso vive" muda.
- O mesmo parágrafo registra que **não existe gate pré-LLM por proporção de código** (issue #154, fechada como wont-do).

**2. A evidência que fechou a #154 é ela própria acoplada ao domínio** — vale registrar, não vale reabrir. O argumento foi que a razão de fences é *anti-correlacionada* com o veredito de extract, porque o chunk de maior razão do corpus (87% fenced) produziu nota: o fence era um bloco Python cujos **comentários enunciavam o conceito**. Essa medição não se transfere para um corpus sem código.

Para este vault (misto, com código de verdade) a conclusão continua valendo e **nada muda**. O registro serve para o caso de o acervo virar majoritariamente prosa: aí a #154 teria sido fechada com evidência que já não se aplica. É um bom lembrete de que decisão medida num corpus tem validade limitada a ele — o mesmo princípio que a ADR-003 e o RUNBOOK aplicam aos thresholds.

**Comment by ricmed** [comment]
Acréscimo de escopo: **um few-shot para trecho narrativo/ficcional.**

O prompt de extract trata narrativa como enchimento. `literature_note.md:39-44` lista sob rejeição:

> Histórias anedóticas sem princípio extraível

A cláusula `sem princípio extraível` tecnicamente permite ficção com conceito. Mas o resto do prompt empurra na direção contrária: existe a categoria de rejeição `narrative`, os quatro exemplos de rejeição (`:344-370`) são todos de preâmbulo/marketing, e **não há nenhum exemplo positivo** de trecho narrativo que renda candidato válido. Na prática o modelo vai rejeitar quase tudo que for prosa ficcional.

Para ficção, a narrativa não é o invólucro — é a carga. Uma cena sobre classificação de cidadãos por nota **é** o conceito, encenado.

Concretamente, acrescentar a `rejection_examples` ou ao lado de `accepted_example` um par que demonstre a distinção:

- **Aceito**: chunk narrativo cuja cena instancia um princípio transferível. Tese extraída do mecanismo encenado (não do enredo), `anchor_quote` literal do texto literário, tags transferíveis (`classificacao_social`, `etica_algoritmica`).
- **Rejeitado**: chunk narrativo que é só enredo — sem princípio destacável da situação particular.

Sem esse par, o modelo não tem como distinguir "narrativa que carrega conceito" de "narrativa que é transição de capítulo", e o default do prompt é rejeitar.

Relacionado: #165 argumenta que o caminho normal para ficção deve ser SRC + ZTL manual, sem processar o livro inteiro. Este few-shot cobre o caso em que uma obra narrativa **é** harvestada — ensaio, biografia, reportagem longa, ou uma ficção que se decida processar. Os dois não se excluem.

**Comment by ricmed** [comment]
## Solução adotada (branch `feat/dominio-configuravel-e-pontes-entre-dominios`)

Few-shots saíram dos prompts e foram para `config/domain_examples.yaml` (os dois domínios lado a lado, de propósito: um "nível 5" de ML ao lado de um de Filosofia ensina densidade conceitual, não proximidade de campo).

- Loader folha `zettel/domain_examples.py` (yaml + pydantic; `accepted_example` valida contra `LiteratureChunkOutput`).
- `render_for_prompt` devolve **todas** as chaves sempre (string vazia se a seção faltar) — `fill_template` deixaria `{relevance_examples}` literal.
- Callers carregam **uma vez por run** e listam as chaves no literal `mapping = {...}` (contrato AST de `tests/test_prompts.py`).
- `fragmented` generalizado no prompt, sem parametrizar o enum.
- Citações de artigo neutralizadas (formato, não julgamento).
- `grep` de jargão de ML em `prompts/` fica só em `ptbr_guard.md:9` (benigno).

ADR-042. Depende de #156 (T1).

**Comment by ricmed** [comment]
Implementado em https://github.com/ricmed/zettel_app/pull/167 (commit `39a003a`, few-shot de ficção em `e0157b4`).

`zettel/domain_examples.py` espelha `taxonomy.py` (Pydantic + loader tipado + `render_for_prompt` que devolve toda chave sempre, nunca deixa `{chave}` cru para o `fill_template`). `accepted_example` valida contra `LiteratureChunkOutput` no load — recupera a cobertura de `test_json_examples_are_valid_json` que se perderia ao tirar o JSON do `.md`. As 6 chaves de `literature_note` e 2 de `permanent_note` entram no literal `mapping = {...}` de `extractor.py`/`connector.py`, respeitando o contrato AST de `test_prompts.py`.

O item que tinha ficado pendente (few-shot de narrativa/ficção) foi resolvido à parte em #166.

Testado: grep de vocabulário ML nos prompts vazio; nenhum placeholder vaza no prompt preenchido.
